### ISSUE TITLE
Label of the current task in the build log in TeamCity sometimes gets mixed up.

### ISSUE THREAD

**Issue by chrhol** [Starcounter/Starcounter#3266]
The label of the current executing task in the build log sometimes gets mixed up with some earlier task making a little harder to read the log. 

In the snippet below, `[sqlregression_test-bat]` comes up both when starting the next test and in the end of the current (as well as starting the test after that)

```
[sqlregression_test-bat] -- Running process "staradmin kill all"
[06:08:43][sqlregression_test-bat] -- Running test "star_loop_test.bat" (max duration: 60 s) at "C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-161\Level1\src\Server\sccli\star_loop_test.bat"
[06:08:43][sqlregression_test-bat] star_loop_test-bat (32s)
[06:08:43][star_loop_test-bat] -- Running process "C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-161\Level1\src\Server\sccli\star_loop_test.bat "
[06:08:43][star_loop_test-bat] Test is going to loop 5 times:
[06:08:44][star_loop_test-bat] 1
[06:08:58][star_loop_test-bat] 2
[06:09:03][star_loop_test-bat] 3
[06:09:08][star_loop_test-bat] 4
[06:09:12][star_loop_test-bat] 5
[06:09:16][star_loop_test-bat] Star.exe loop tests finished successfully!
[06:09:16][star_loop_test-bat] 06:09:16.8868233: process 'scservice' successfully killed!
[06:09:16][star_loop_test-bat] 06:09:16.8888234: process 'scipcmonitor' successfully killed!
[06:09:16][star_loop_test-bat] 06:09:16.8898235: process 'scnetworkgateway' successfully killed!
[06:09:16][star_loop_test-bat] 06:09:16.8898235: process 'scadminserver' successfully killed!
[06:09:16][star_loop_test-bat] 06:09:16.8908236: process 'sccode' successfully killed!
[06:09:16][star_loop_test-bat] 06:09:16.8918236: process 'scdata' successfully killed!
[06:09:16][star_loop_test-bat] 06:09:16.8918236: process 'scdblog' successfully killed!
[06:09:16][star_loop_test-bat] 06:09:16.8928237: process 'scsqlparser' successfully killed!
[06:09:16][star_loop_test-bat] 06:09:16.8938237: process 'sctrayicon' successfully killed!
[06:09:16][sqlregression_test-bat] -- Running process "staradmin kill all"
[06:09:16][sqlregression_test-bat] -- Running test "dbmapping_test.bat" (max duration: 60 s) at "C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-161\Level1\src\Starcounter.Extensions.Tests\dbmapping_test.bat"
[06:09:16][sqlregression_test-bat] dbmapping_test-bat (5m:06s)
[06:09:16][dbmapping_test-bat] -- Running process "C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-161\Level1\src\Starcounter.Extensions.Tests\dbmapping_test.bat "
[06:09:16][dbmapping_test-bat] 
[06:09:16][dbmapping_test-bat] C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-161\Level1\src\Starcounter.Extensions.Tests>IF "True" == "False" GOTO :EOF 
```

Another snippet from the same build, when generating installer and uploading package.

```
dbmapping_test-bat] Succeeded generating unique installer!
[06:14:21][dbmapping_test-bat] ---------------------------------------------------------------
[06:14:21][dbmapping_test-bat] Copying built Level0 binaries.
[06:14:21][dbmapping_test-bat] ---------------------------------------------------------------
[06:14:21][dbmapping_test-bat] Starting tool: 'Level0 Build'
[06:14:21][dbmapping_test-bat] Uploading Level0 'develop' build artifacts to mapped FTP directory: 'Z:\SCBuilds\Level0\develop\LatestStable'
[06:14:22][dbmapping_test-bat] Updating Error Help Pages on Github wiki.
[06:14:22][dbmapping_test-bat] Skipping updating error pages...
[06:14:22][dbmapping_test-bat] Switching back to build output directory.
[06:14:23][dbmapping_test-bat] Process exited with code 0
```

The log comes from 
https://teamcity.starcounter.org/viewLog.html?buildId=126&tab=buildResultsDiv&buildTypeId=Starcounter_DevelopNightly

**Comment by alemoi** [comment]
I will try to fix that by putting a sleep between tests :) Because i simply do the Console.WriteLine, and its up to TeamCity to pick them up correctly.

**Comment by alemoi** [comment]
I just pushed a possible fix with sleeps via 2d5d1696e7687163822d8157354ea2b70fbd52c6
Let me know how it goes!

**Comment by chrhol** [comment]
> I will try to fix that by putting a sleep between tests :) Because i simply do the Console.WriteLine, and its up to TeamCity to pick them up correctly.

Aha, so it's teamcity that messes something up. 

It looks better now I think. There were some mixups even after your change, but I made a few minor changes in the dbmapping_test script, and now the mixup is gone. Maybe the script did something that made TeamCity unhappy.

Anyways, this is a minor issue and this work good enough :) I will close it.
